source/classes/Westley.js

```diff
@@ -58,24 +41,42 @@
 			throw new Error("Invalid command");
 		}
 		var commandComponents = commandTools.extractCommandComponents(command),
-			commandKey = commandComponents.shift();//,
-			//commandFilename = "command." + commandKey + ".js",
-			//commandExecute;
-		if (!availableCommands.hasOwnProperty(commandKey)) {
-			throw new Error("Unrecognised command: " + commandKey);
-		}
-		// try {
-		// 	commandExecute = require("__buttercup/classes/commands/" + commandFilename);
-		// } catch (err) {
-		// 	throw new Error("Unrecognised command: " + commandKey);
-		// }
-		var commandToExecute = availableCommands[commandKey];
+			commandKey = commandComponents.shift();
+
+		var commandObject = this._getCommandForKey(commandKey);
 		this._history.push(command);
-		commandToExecute.apply(commandToExecute, [this._dataset].concat(commandComponents));
+		commandObject.execute.apply(commandObject, [this._dataset].concat(commandComponents));
 		return this;
 	};
 
 	/**
+	 * Gets a command by its key from the cache with its dependencies injected
+	 * @param {String} commandKey The key of the command
+	 * @returns {Command} Returns the command
+	 * @memberof Westley
+	 */
+	Westley.prototype._getCommandForKey = function(commandKey) {
+		// If the command doesn't exist in the cache
+		if (this._cachedCommands[commandKey] === undefined) {
+			// Get the command object and inject its dependencies
+			var requirement = new (require("__buttercup/classes/commands/command." + commandKey + ".js"))();
+
+			if (requirement.injectSearching !== undefined) {
+				requirement.injectSearching(searching);
+			}
+
+			if (requirement.injectEntry !== undefined) {
+				requirement.injectEntry(entry);
+			}
```

Comment:
It's not so much about removing the dependency (it's fine to have the dependency) but rather about changing the control of how it gets that dependency.

The commands themselves need a `searching` module, but it's not their responsibility to get it. They should just be provided it.

From a practical point of view, this means that we can fake out the behaviour of the dependencies in tests. If you take a look at the tests I've written so far, you can see where I am, for example, faking the `searching` module to make sure that an error is thrown when it can't find the module.